### PR TITLE
Posts

### PR DESCRIPTION
**Pull request by muskanbagrecha**
1. Added posts functionality with like bookmark comment. 
2. Added explore page with filters.

### CODE REVIEW

File: src/router/AppRouter.tsx [hunk 1]
```diff
@@ -1,10 +1,36 @@
...
-import { Homepage, Profilepage } from "../pages";
+import {
+  Homepage,
+  Profilepage,
+  Bookmarkpage,
+  Likespage,
+  Loginpage,
+  Signuppage,
+  Explorepage,
+  // Otherprofilepage,
```
Comment (zoef-yt) [inline src/router/AppRouter.tsx:10]: remove commented code.

File: src/pages/profilepage/components/your-followers/YourFollowers.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        <div>
+          {followersList?.map((user: { id: string }) => {
+            const currentUser = allUsers?.find((profile: any) => {
+              return profile.uid === user.id;
+            });
+            const { photoURL, displayName, uid } = currentUser ?? {
+              photoURL: "",
+              displayName: "",
+              uid: "",
+            };
```
Comment (zoef-yt) [inline src/pages/profilepage/components/your-followers/YourFollowers.tsx:22]: Nice way of handling this.

File: src/pages/profilepage/components/main-profile/MainProfile.tsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+          <span>Followers</span>
+        </div>
+        <div className="flex flex-col items-center  cursor-pointer">
+          <span>{followings}</span>
+          <span>Following</span>
+        </div>
+        <div className="flex flex-col items-center  cursor-pointer">
+          <span>{posts.length}</span>
+          <span>Posts</span>
+        </div>
```
Comment (zoef-yt) [inline src/pages/profilepage/components/main-profile/MainProfile.tsx:205]: You can make them into one component.

File: src/pages/profilepage/components/main-profile/MainProfile.css [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    position: relative;
+    color: var(--color-current);
+}
+
+.profile-overlay{
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    transition: .5s ease;
```
Comment (zoef-yt) [inline src/pages/profilepage/components/main-profile/MainProfile.css:11]: Do add which property you want to apply transition to

File: src/pages/homepage/components/post/Post.css [hunk 5]
```diff
@@ -2,14 +2,40 @@
...
+    min-height: min-content;
+}
+
 .post{
-    min-height: 20rem;
+    animation-name: post-card-animation;
+	animation-duration: 0.4s;
+}
+
+@keyframes post-card-animation {
```
Comment (zoef-yt) [inline src/pages/homepage/components/post/Post.css:14]: Nice subtle animation.

File: src/pages/homepage/components/main-feed/MainFeed.css [hunk 6]
```diff
@@ -1,14 +1,3 @@
...
-
-@media screen and (max-width: 640px) {
-    .main-feed{
-        
-    }
-}
\ No newline at end of file
+/* .main-feed{
+    max-width: 300px;
+} */
```
Comment (zoef-yt) [inline src/pages/homepage/components/main-feed/MainFeed.css:3]: remove commented code.

File: src/firebase/firebase-firestore.ts [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+const getCommentsFromPost = async (postId:string) => {
+	try {
+		const postCollection = query(collection(db, `posts/${postId}/comments`), orderBy('timestamp', 'desc'));
+		const allComments = await getDocs(postCollection);
+		const allCommentsData : any[] = [];
+		allComments.forEach((doc) => {
+			console.log(doc.data())
+			allCommentsData.push(doc.data());
+		});
+		// console.log(allCommentsData)
```
Comment (SinchanaMS) [inline src/firebase/firebase-firestore.ts:257]: Remove this log.

File: src/firebase/firebase-firestore.ts [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+		console.log(error);
+	}
+};
+
+
+const editUserProfileImage = async (userId:string, imageFile:File|null) => {
+	if (imageFile) {
+		try {
+			const url = await uploadImageGetUrl(imageFile, `users/${userId}`);
+			console.log(url);
```
Comment (SinchanaMS) [inline src/firebase/firebase-firestore.ts:290]: Remove the log here too.

File: src/pages/homepage/components/comment-section/CommentSection.css [hunk 9]
```diff
@@ -0,0 +1,12 @@
+.comment-holder{
+    position:relative;
+    max-height: 14rem;
+    height: 100%;
+    overflow-y:auto;
+
```
Comment (zoef-yt) [inline src/pages/homepage/components/comment-section/CommentSection.css:6]: remove extra line.

**Review by zoef-yt — APPROVED** [review]
Really great work with the app. gave some suggestions you can implement. Really love how the UI turned out. Loved the small animations and transition great work with the Typescript, redux, and firebase. The mobile version also looks really great. The app works so smoothly with all the functionality great work musk you are good to merge.

**Review by SinchanaMS — APPROVED** [review]
Love how neat the code is! Very self-explanatory. Great work!!

**Pull request merged**